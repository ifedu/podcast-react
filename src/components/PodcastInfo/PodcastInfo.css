.PodcastInfo {
  box-shadow: 0 1px 2px 2px #ddd;
  font-size: 14px;
  width: 300px;
}

.PodcastInfo .section  {
  margin: 8px;
  padding: 12px;
}

.PodcastInfo span {
  font-weight: bold;
}

.PodcastInfo .picture {
  display: flex;
  justify-content: center;
  padding: 30px;
}

.PodcastInfo .picture img {
  width: 200px;
}

.PodcastInfo .title {
  border-bottom: 1px solid #ddd;
  border-top: 1px solid #ddd;
}

.PodcastInfo .title span {
  font-size: 16px;
}

.PodcastInfo .description div {
  font-style: italic;
}
