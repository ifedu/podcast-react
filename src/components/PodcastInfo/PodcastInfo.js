import React from 'react';
import { Link } from 'react-router-dom';

import './PodcastInfo.css';

export function PodcastInfo({ info, song }) {
  return (
    <div className='PodcastInfo'>
      <Link to={`/podcast/${info.collectionId}`} state={{ song }}>
        <div className='section picture'><img src={info.artworkUrl600} alt='All Songs Considered'/></div>

        <div className='section title'>
          <span>{info.collectionName}</span>
          <div>by {info.artistName}</div>
        </div>
      </Link>

      <div className='section description'>
        <span>Description</span>
        <div>{song.summary.label}</div>
      </div>
    </div>
  );
}
