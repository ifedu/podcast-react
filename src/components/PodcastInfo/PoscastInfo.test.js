import { cleanup, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import renderer from 'react-test-renderer' ;

import { PodcastInfo } from './PodcastInfo';

const getContent = (pathname = '/') => {
  const props = {
    info: {
      collectionId: 1,
    },
    song: {
      summary: {},
    },
  };

  return <MemoryRouter initialEntries={[{ pathname }]}>
    <Routes>
      <Route path='/podcast/1' element={
        <div data-testid='podcast'><PodcastInfo {...props}/></div>
      }></Route>

      <Route path='/podcast/1/episode/1' element={
        <div data-testid='episode'><PodcastInfo {...props}/></div>
      }></Route>
    </Routes>
  </MemoryRouter>;
}

describe('PodcastInfo', () => {
  const view = getContent('/podcast/1/episode/1');

  afterEach(cleanup);

  it('renders correctly', () => {
    const tree = renderer.create(view).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the podcast info', async() => {
    render(view);

    await waitFor(() => {
      const podcast = screen.queryByTestId('podcast-info');
      expect(podcast).toBeInTheDocument();
    });
  });

  it('navigates from episode page to the podcast page', async () => {
    render(view);

    await waitFor(() => {
      const podcastInfo = screen.queryByTestId('episode');
      expect(podcastInfo).toBeInTheDocument();
    });

    let link = screen.getByAltText('All Songs Considered');
    userEvent.click(link);

    await waitFor(() => {
      const episode = screen.queryByTestId('podcast');
      expect(episode).toBeInTheDocument();
    });
  });
});
