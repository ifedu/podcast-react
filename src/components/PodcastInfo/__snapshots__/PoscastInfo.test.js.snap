// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastInfo renders correctly 1`] = `
<div
  data-testid="episode"
>
  <div
    className="PodcastInfo"
    data-testid="podcast-info"
  >
    <a
      href="/podcast/1"
      onClick={[Function]}
    >
      <div
        className="section picture"
      >
        <img
          alt="All Songs Considered"
        />
      </div>
      <div
        className="section title"
      >
        <span />
        <div>
          by 
        </div>
      </div>
    </a>
    <div
      className="section description"
    >
      <span>
        Description
      </span>
      <div />
    </div>
  </div>
</div>
`;
